bundles/framework/printout/view/BasicPrintout.js

```diff
@@ -598,36 +598,43 @@ Oskari.clazz.define('Oskari.mapframework.bundle.printout.view.BasicPrintout',
         },
 
         /**
-         * @private @method openPostURLinWindow
+         * @private @method openPostPrint
          * Sends the gathered map data to the server to save them/publish the map.
          *
-         * @param {String} printUrl Url to print service action route GetPrint
-         * @param {String} geoJson Stringified GeoJSON
-         * @param {String} tileData Stringified tile data
-         * @param {String} customStyles Stringified styles
+         * @param {String} printUrl
+         * @param {Object} params
+         * @param {Object} customStyles
          *
          */
-        openPostURLinWindow: function (printUrl, geoJson, tileData, tableData, customStyles) {
-            var me = this;
-            var wopParm = this._getWindowSpecs();
-            me.mainPanel.find('#oskari_print_formID').attr('action', printUrl);
-            me.mainPanel.find('input[name=customStyles]').val(customStyles);
-            // Are these used anymore??
-            if (geoJson) {
-                // UTF-8 Base64 encoding
-                var textu8 = unescape(encodeURIComponent(geoJson));
-                me.mainPanel.find('input[name=geojson]').val(jQuery.base64.encode(textu8));
-            }
-            if (tileData) {
-                me.mainPanel.find('input[name=tiles]').val(tileData);
+        getPostPrint: function (printUrl, params, customStyles) {
+            const payload = { customStyles };
+            let fileName = params.pageTitle || 'print';
+            const format = FORMAT_OPTIONS.find(format => format.mime === params.format);
+            if (format) {
+                fileName += '.' + format.name;
             }
-            if (tableData) {
-                me.mainPanel.find('input[name=tabledata]').val(tableData);
-            }
-
-            window.open('about:blank', 'map_popup_111', wopParm);
-
-            me.mainPanel.find('#oskari_print_formID').submit();
+            const showSpinner = visible => this.instance.getSandbox().postRequestByName('ShowProgressSpinnerRequest', [visible]);
+            showSpinner(true);
+            const successCb = blob => {
+                if (navigator.msSaveOrOpenBlob) {
+                    navigator.msSaveOrOpenBlob(blob, fileName);
+                } else {
+                    const url = window.URL.createObjectURL(blob);
+                    const elem = document.createElement('a');
+                    elem.href = url;
+                    elem.download = fileName;
+                    document.body.appendChild(elem);
+                    elem.click();
+                    document.body.removeChild(elem);
+                    window.URL.revokeObjectURL(url);
+                }
+                showSpinner(false);
+            };
+            const errorCb = error => {
+                Oskari.log('BasicPrintout').error(error);
```

Comment:
Should we give the user some friendlier error message? Now just the spinner stops and dev-console gets a message.